packages/style-value-parser/src/css-types-from-tokens/calc.js

```diff
@@ -186,3 +189,53 @@ export class Calc {
       .map(([_, value, _closeParen]) => new Calc(value));
   }
 }
+
+function calcValueToString(value: CalcValue | string): string {
+  // plain numbers
+  if (typeof value === 'number') {
+    return value.toString();
+  }
+
+  // raw identifiers/constants
+  if (typeof value === 'string') {
+    return value;
+  }
+
+  // explicit user‑written group
+  if (value != null && typeof value === 'object' && 'expr' in value) {
+    const group: Group = value as any;
+    return '(' + calcValueToString(group.expr) + ')';
+  }
+
+  // binary operations
+  if (
+    value != null &&
+    typeof value === 'object' &&
+    'left' in value &&
+    'right' in value &&
+    typeof (value as any).type === 'string'
+  ) {
+    const opNode: Addition | Subtraction | Multiplication | Division =
+      value as any;
```

Comment:
This should also work if you've already checked for all non-object types previously.

```suggestion
  if (
    value != null &&
    typeof value === 'object' &&
    ( value.type === '+' ||
      value.type === '-' ||
      value.type === '*' ||
      value.type === '/' )
  ) {
```